docs/user-guide/nginx-configuration/annotations.md

```diff
@@ -129,6 +130,8 @@ In some cases, you may want to "canary" a new set of changes by sending a small
 
 * `nginx.ingress.kubernetes.io/canary-by-header-value`: The header value to match for notifying the Ingress to route the request to the service specified in the Canary Ingress. When the request header is set to this value, it will be routed to the canary. For any other header value, the header will be ignored and the request compared against the other canary rules by precedence. This annotation has to be used together with . The annotation is an extension of the `nginx.ingress.kubernetes.io/canary-by-header` to allow customizing the header value instead of using hardcoded values. It doesn't have any effect if the `nginx.ingress.kubernetes.io/canary-by-header` annotation is not defined.
 
+* `nginx.ingress.kubernetes.io/canary-match-strategy`: The match strategy to allow using specific strategy to match the header value. If the strategy is not defined, the default strategy `exact match` will be used, the `nginx.ingress.kubernetes.io/canary-by-header` annotation doesn't have any affect if the strategy is not implemented. Now the available strategies are `exact`、`regex`.
```

Comment:
> the `nginx.ingress.kubernetes.io/canary-by-header` annotation doesn't have any affect if the strategy is not implemented. 

Do you mean `canary-match-strategy` has no effect on `canary-by-header`? Instead of specifying what this is not impacting, I'd say something around "The match strategy is useful only with `canary-by-header-value` annotation".

Reply:
I'm so sorry i didn't make myself clear, i mean it will match failed when the pattern is not be implemented.

Reply:
done